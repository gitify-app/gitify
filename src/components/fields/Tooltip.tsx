import { QuestionIcon } from '@primer/octicons-react';
import { type FC, type ReactNode, useState } from 'react';

export interface ITooltip {
  name: string;
  tooltip: ReactNode | string;
}

export const Tooltip: FC<ITooltip> = (props: ITooltip) => {
  const [showTooltip, setShowTooltip] = useState(false);

  return (
    <span
      id={props.name}
      aria-label={props.name}
      className="relative"
      onMouseEnter={() => setShowTooltip(true)}
      onMouseLeave={() => setShowTooltip(false)}
    >
      <QuestionIcon className="text-blue-500 ml-1" />
      {showTooltip && (
        <div className="absolute bg-white dark:bg-gray-sidebar border border-gray-300 rounded p-2 shadow">
          <div className="text-gray-700 dark:text-white text-xs text-left">
            {props.tooltip}
          </div>
        </div>
      )}
    </span>
  );
};
