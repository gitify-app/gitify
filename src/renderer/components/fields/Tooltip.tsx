import { QuestionIcon } from '@primer/octicons-react';
import { Box } from '@primer/react';
import { type FC, type ReactNode, useState } from 'react';

export interface ITooltip {
  name: string;
  tooltip: ReactNode | string;
}

export const Tooltip: FC<ITooltip> = (props: ITooltip) => {
  const [showTooltip, setShowTooltip] = useState(false);

  return (
    <Box
      id={props.name}
      aria-label={props.name}
      className="relative"
      onMouseEnter={() => setShowTooltip(true)}
      onMouseLeave={() => setShowTooltip(false)}
      data-testid={`tooltip-${props.name}`}
    >
      <QuestionIcon className="text-gitify-tooltip-icon" />
      {showTooltip && (
        <Box className="absolute left-[-80px] z-10 w-60 rounded-sm border border-gray-300 p-2 shadow-sm bg-gitify-tooltip-popout">
          <Box className="text-left text-xs text-gitify-font">
            {props.tooltip}
          </Box>
        </Box>
      )}
    </Box>
  );
};
