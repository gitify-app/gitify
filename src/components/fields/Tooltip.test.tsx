import { fireEvent, render, screen } from '@testing-library/react';

import * as TestRenderer from 'react-test-renderer';

import { type ITooltip, Tooltip } from './Tooltip';

describe('components/fields/Tooltip.tsx', () => {
  const props: ITooltip = {
    name: 'tooltip',
    tooltip: 'This is some tooltip text',
  };

  it('should render', () => {
    const tree = TestRenderer.create(<Tooltip {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should display on mouse enter / leave', () => {
    render(<Tooltip {...props} />);

    const tooltipElement = screen.getByLabelText('tooltip');

    fireEvent.mouseEnter(tooltipElement);
    expect(tooltipElement).toMatchSnapshot();

    fireEvent.mouseLeave(tooltipElement);
    expect(tooltipElement).toMatchSnapshot();
  });
});
