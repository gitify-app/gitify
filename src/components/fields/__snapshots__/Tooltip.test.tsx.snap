// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/fields/Tooltip.tsx should display on mouse enter / leave 1`] = `
<span
  aria-label="tooltip"
  class="relative"
  id="tooltip"
>
  <svg
    aria-hidden="true"
    class="text-blue-500 ml-1"
    fill="currentColor"
    focusable="false"
    height="16"
    style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
    viewBox="0 0 16 16"
    width="16"
  >
    <path
      d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
    />
  </svg>
  <div
    class="absolute bg-white dark:bg-gray-sidebar border border-gray-300 rounded p-2 shadow"
  >
    <div
      class="text-gray-700 dark:text-white text-xs text-left"
    >
      This is some tooltip text
    </div>
  </div>
</span>
`;

exports[`components/fields/Tooltip.tsx should display on mouse enter / leave 2`] = `
<span
  aria-label="tooltip"
  class="relative"
  id="tooltip"
>
  <svg
    aria-hidden="true"
    class="text-blue-500 ml-1"
    fill="currentColor"
    focusable="false"
    height="16"
    style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
    viewBox="0 0 16 16"
    width="16"
  >
    <path
      d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
    />
  </svg>
</span>
`;

exports[`components/fields/Tooltip.tsx should render 1`] = `
<span
  aria-label="tooltip"
  className="relative"
  id="tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <svg
    aria-hidden="true"
    className="text-blue-500 ml-1"
    fill="currentColor"
    focusable="false"
    height={16}
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
    />
  </svg>
</span>
`;
